geoips/interfaces/text_based/ascii_palettes.py

```diff
@@ -0,0 +1,181 @@
+        )
+    carray = numpy.zeros([len(lines), 3])
+    for num, line in enumerate(lines):
+        color_nums = line.strip().split()
+        if len(color_nums) != 3:
+            raise AsciiPaletteError(
+                f"One or more lines of the ascii palette in {fpath} are missing "
+                "one or more values of the 'rgb' triplet. Please fix."
+            )
+        try:
+            carray[num, :] = [float(val) for val in line.strip().split()]
+        except ValueError as e:
+            raise AsciiPaletteError(
+                f"Invalid ascii palette found in {fpath}. See resulting error: {e}."
+            )
+
+    # Normalize from 0-255 to 0.0-1.0
+    if carray.max() > 1.0:
+        carray /= 255.0
+
+    # Test to be sure all color array values are between 0.0 and 1.0
+    if not (carray.min() >= 0.0 and carray.max() <= 1.0):
+        raise AsciiPaletteError("All values in carray must be between 0.0 and 1.0.")
+
+    if reverse is True:
+        carray = numpy.flipud(carray)
+
+    if cmap_name is not None:
+        cmap_name = basename(fpath)
+    cmap = colors.ListedColormap(carray, name=cmap_name)
+    return cmap
+
+
+class AsciiPalettesPlugin(BaseTextPlugin):
+    """Class representing an Ascii Palette Used to Construct a GeoIPS Colormapper."""
+
+    @property
+    def colormap(self):
+        """Ascii Palette derived from 'self.plugin_entry'.
+
+        Returns
+        -------
+        cmap: Matplotlob.colors.ListedColormap
+            - The colormap derived from the ascii palette
+        """
+        if not hasattr(self, "_colormap"):
+            path_to_ascii = str(
+                files(self.plugin_entry["package"]) / self.plugin_entry["relpath"]
+            )
+            self._colormap = from_ascii(path_to_ascii)
+        return self._colormap
+
+    def __call__(self, **kwargs):
+        """Generate a dictionary of colormap information used for linear norm cmaps.
+
+        This information used in both colorbar and image production throughout
+        GeoIPS image output specifications.
+
+        Parameters
+        ----------
+        kwargs: dict of arguments
+            - For a full list of arguments that will be provided to this function, see:
+              geoips.plugins.colormappers.ascii_based:call
+
+        Returns
+        -------
+        mpl_colors_info : dict
+            * Specifies matplotlib Colors parameters for use in both plotting and
+              colorbar generation
+
+        See Also
+        --------
+        :ref:`api`
+            See geoips.image_utils.mpl_utils.create_colorbar for field descriptions.
+        """
+        min_val = None
+        max_val = None
+        kwargs = SimpleNamespace(**kwargs)
+        if kwargs.data_range is not None:
+            min_val = kwargs.data_range[0]
+            max_val = kwargs.data_range[1]
+
+        mpl_cmap = self.colormap
+
+        LOG.info("Setting norm")
+
+        mpl_norm = colors.Normalize(vmin=min_val, vmax=max_val)
```

Comment:
I think we may want to make this more general. [There are a lot of different normalization algorithms that we could provide here](https://matplotlib.org/stable/api/colors_api.html). We might want to implement some of these other colormappers in the future, so it would be good to make the normalization type a keyword argument.

Reply:
This has been generalized now. See ``geoips.plugins.modules.colormappers.ascii_based``